R/dist_spec.R

```diff
@@ -343,42 +384,49 @@ max.dist_spec <- function(x, ...) {
 #'
 #' # The maxf the sum of two distributions
 #' discretise(dist1 + dist2, strict = FALSE)
-discretise <- function(x, strict = TRUE) {
+discretise.dist_spec <- function(x, strict = TRUE, ...) {
   if (!is(x, "dist_spec")) {
```

Comment:
This should now be converted to a `check_dist_spec()` function + another for `multi_dist_spec()` as they are repeated a lot here.

Reply:
I would gently push back on this as the only repeated thing is the class check so I think having a separate function adds is overkill here.